src/pages/settings/Settings.tsx

```diff
@@ -0,0 +1,145 @@
+const themesToSave: ValidColorSchemeName[] = ["predefined", "dark", "light"]
+
+/**
+ * Settings Page
+ */
+export const SettingsPage: SettingsStackScreen<"Settings"> = props => {
+    const user = props.route.params.user
+
+    const context = useContext(SettingsContext)
+
+    const settings = context.settings
+    const setSettings = context.setSettings
+
+    //App state theme
+    const theme = settings.theme
+    //Settings setter
+
+    //RadioButtonGroup theme state and setter
+    const [selectedTheme, setSelectedTheme] =
+        useState<ValidColorSchemeName>(theme)
+
+    //RadioButtonGroup carriere state and setter
+    const [selectedMatricola, setSelectedMatricola] = useState(
+        user.carriere[0].matricola
+    )
+
+    //for testing logged in/out view
+    const [logged, setLogged] = useState(false)
+
+    const [isModalVisible, setModalVisible] = useState(false)
+
+    const { navigate } = useNavigation()
+
+    const settingsList: SettingOptions[] = [
+        {
+            title: "Aspetto",
+            subtitle: "Dark, light mode",
+            icon: settingsIcons.modify,
+            callback: () => {
+                setModalVisible(true)
+            },
+        },
+        {
+            title: "Aiuto",
+            subtitle: "Centro assistenza, contattaci, informativa privacy",
+            icon: settingsIcons.help,
+            callback: () => {
+                navigate("Help")
+            },
+        },
+        { title: "Disconnetti", icon: settingsIcons.disconnect },
+    ]
+
+    return (
+        <View style={{ flex: 1 }}>
+            <SettingsScroll title="Impostazioni">
+                {logged ? (
+                    <UserDetailsTile
+                        codPersona={user.codPersona}
+                        profilePic={user.profilePic}
+                        nome={user.nome}
+                        cognome={user.cognome}
+                        onPress={() => setLogged(false)}
+                    />
+                ) : (
+                    <UserAnonymousTile
+                        showRipple={false}
+                        onLogin={() => setLogged(true)}
+                    />
+                )}
+                {logged && (
+                    <View>
+                        {user.carriere?.map((carriera, index) => {
+                            return (
+                                <CourseTile
+                                    key={index}
+                                    matricola={carriera.matricola}
+                                    type={carriera.type}
+                                    selected={
+                                        selectedMatricola === carriera.matricola
+                                    }
+                                    onPress={() =>
+                                        setSelectedMatricola(carriera.matricola)
+                                    }
+                                />
```

Comment:
scusa questa mi era passata di mente, forse mi ero addirittura dimenticato di dirtelo 😬 
abbiamo deciso per conformità di fare una lieve modifica di design, il selettore della matricola non deve essere più quei 3 radio button ma un modal con i radio button dentro (identico al selettore del color theme)
Praticamente pianta lì un tasto "Cambia matricola" uguale a quello "Login" che apre un modal come quello del tema.

Reply:
Ok, ho fatto il cambiamento. Nel modal non sapevo se aggiungere anche il tipo di matricola oltre il numero. In caso si può cambiare, per il momento mostra solo le matricole.

Reply:
ma si dai infilaci anche il tipo di matricola, è più intuitivo

Reply:
poi sai cosa, Course come nome per la carriera non mi sembra molto giusto, in webeep-sync course lo uso per i corsi, e boh confusionario, solo che ho visto anche sui servizi online non c'è un nome comune in inglese, sul portale viene chiamato "ID" o "Identification Number", e nelle API del poli è semplicemente chiamato "matricola" so idk 🤷‍♂️ 
se trovi un nome migliore cambialo se no boh lascialo lì e ci pensiamo un po' su finché non ci tocca fare qualcosa con i corsi effettivamente lol

Reply:
Okay, eh si il nome in effetti è ingannevole. Forse Career o UniCareer sarebbe meglio.

Reply:
si decisamente più chiaro

Reply:
Per non duplicare il codice ho creato un componente CareerColumn (colonna con matricola e tipo matricola) che viene utilizzato sia nella CareerTile (la riga con il bottone Cambia Matricola) sotto l'avatar, sia come children nelle SelectTile dentro il modale per la scelta della matricola.

Reply:
Mi è sorto un dubbio. Le matricole teoricamente possono essere qualsiasi numero? perchè se diventano più di tre non ci stanno più nel wrapper dentro il modale. Forse sarebbe il caso di aggiustare la dimensione del modale programmaticamente in base al numero di scelte o di inserire una scrollview dentro il modale.

Reply:
Per il momento ho fatto che l'altezza aumenta per mostrare tutto il contenuto necessario

Reply:
sì decisamente scelta giusta, dovrebbe assolutamente aumentare con il contenuto